dnsutil/util.go

```diff
@@ -57,8 +59,12 @@ func TrimDomainName(s, origin string) string {
 		return strings.TrimSuffix(s, origin)
 	}
 
+	original_s := s
+	s = AddOrigin(s, ".")
```

Comment:
you prolly can't upcall to dns.Fqdn() which is basically what this does?

Reply:
Done